### DISCUSSION TITLE
Help needed with Trocador plugin update to 2.0 (2.0.3)

### DISCUSSION THREAD

**Discussion by GergelyGombai** [btcpayserver/btcpayserver#6417] [Plugins & integrations] [answered]
Hello! @NicolasDorier I was told to mention you, as you might be able to help me with my problem.

I changed the version to the said, 2.0.3 version, and the .net version to 8 instead of 6.

When I try to build the plugin, I get:

/Users/comradeblin/btcpay/trocador-plugin/BTCPayServer.Plugins.Trocador/TrocadorController.cs(57,49): error CS1061: 'GenericPaymentMethodData' does not contain a definition for 'Value' and no accessible extension method 'Value' accepting a first argument of type 'GenericPaymentMethodData' could be found (are you missing a using directive or an assembly reference?) [/Users/comradeblin/btcpay/trocador-plugin/BTCPayServer.Plugins.Trocador/BTCPayServer.Plugins.Trocador.csproj]

/Users/comradeblin/btcpay/trocador-plugin/BTCPayServer.Plugins.Trocador/TrocadorController.cs(59,55): error CS1061: 'GenericPaymentMethodData' does not contain a definition for 'Key' and no accessible extension method 'Key' accepting a first argument of type 'GenericPaymentMethodData' could be found (are you missing a using directive or an assembly reference?) [/Users/comradeblin/btcpay/trocador-plugin/BTCPayServer.Plugins.Trocador/BTCPayServer.Plugins.Trocador.csproj]

as errors, and 11 warnings.

Any ideas what new methods I need to use?

Thanks in advance!

**Comment by GergelyGombai** [comment]
Hello guys! It looks like I was able to "fix" the errors and build the plugin without any. But there are other things:

1. When I fix the warnings about the Singleton - AddUIExtension and the ActivePageClass, the Trocador plugin just does not show up at all on the sidebar(therefore does not work) - This I can probably fix, as I just realized I used the wrong order of strings
2. On the invoice, the primary currency's details (QR, address) don't show up. (therefore users cannot pay by it) - image attached
3. The Trocador/Altcoins text just hopped on the side, even though it should be centered OR next to Monero. - image attached
4. The NoScript page leads me to a 404.

![telegram-cloud-photo-size-4-5931654778910262466-y](https://github.com/user-attachments/assets/32f53d3f-e86d-44a9-94d0-08b8d2aeb3e5)

![telegram-cloud-photo-size-4-5931654778910262467-y](https://github.com/user-attachments/assets/4d07ba14-42b6-4280-9f53-8d34dd2b8dc7)

**Reply by dennisreimann** [reply]
For 1. you should use it like this: `services.AddUIExtension("header-nav", "MyNavExtension");`

For 2. and 3.: Is there a public page where I can take a look at the checkout view and see the source code?

**Reply by GergelyGombai** [reply]
Hey! Thanks! Funny you should comment, I just managed to fix both of them. Now only the 3. and 4. is left. :D

**Reply by dennisreimann** [reply]
For 3.: Is there a public page where I can see it? You can also ping me on  Mattermost.

**Comment by GergelyGombai** [comment]
Does anyone have any ideas? I'd really appreciate the help.

**Comment by NicolasDorier** [comment, marked as answer]
@GergelyGombai 

> The NoScript page leads me to a 404.

IMHO, I don't think you should care about it, probably remove it.

>  AddUIExtension and the

Be careful about this the arguments are inversed compared as before. If it was `services.Add<UIExtension>(a,b)`, it is `services.AddUIExtensions(b,a)`.

You can ping me on chat.btcpayserver.org

**Comment by NicolasDorier** [comment]
@GergelyGombai I can help you migrate, do you have a PR open somewhere with your work in progress?

Linking issue on their repo as well: https://github.com/rafael-xmr/trocador-plugin/issues/7

**Reply by GergelyGombai** [reply]
It's been already done for like a week now. I just forgot to close this discussion, sorry. Also, the repo is no longer rafael's. It's mine. Rafael no longer works at Trocador.

**Reply by GergelyGombai** [reply]
Correction: The third is not done. So if you have a suggestion for that I'd gladly accept it.

**Reply by NicolasDorier** [reply]
@GergelyGombai can you then ask github support to remove the "forked from https://github.com/rafael-xmr/trocador-plugin"

That way people can post issues on the repo and use the search feature.

**Comment by GergelyGombai** [comment]
Sure. As soon as arrive home. Thanks for letting me know.

**Reply by dennisreimann** [reply]
@GergelyGombai I've added GergelyGombai/trocador-plugin#1 with some UI updates and also a fix of the problem mentioned above.
